### ISSUE TITLE
Currently killing on traffic instead of open TCP connections

### ISSUE THREAD

**Issue by hexylena** [erasche/gie-proxy#1]

**Comment by hexylena** [comment]
I did some small testing today...:
- set up a small server that just echoed 'hello, world' 
- sat and watched the `netstat` output under various conditions

For a static HTML page, the connection closes quite quite quickly (a minute or two) whenever firefox decides it isn't going to re-use that TCP connection.

(Websocket servers regularly send keep-alive ping/pong messages which means it really isn't an issue for IPython/Jupyter.)

So what does this mean? Since the browser will close the TCP connection even before the user stops viewing the web page... I _think_ it means we're fine to watch traffic and kill after (say 10 minutes of) inactivity, but I'd like your opinion on this @bgruening

**Comment by hexylena** [comment]
@bgruening indeed, this seems to be how Jupyter/tmpnb works: https://github.com/jupyter/tmpnb/blob/master/spawnpool.py#L389 they also kill "zombie containers" based on their last activity.

**Comment by hexylena** [comment]
Also, @bgruening: https://github.com/minrk/nbbot neat! Might be good for testing.

**Comment by bgruening** [comment]
@erasche my entire concern is that if you leave for a meeting or for lunch and the container is killed because of inactivity we loose data. For IPython/Jupyter this is not a problem they saving the notebooks regularly to prevent to much data lost. We can not do this (currently) it would mess up our history.

**Comment by hexylena** [comment]
Yes, and what I'm saying is "that will already happen under nodejs."
Detecting on traffic is "close enough" to detecting on TCP connections, and
even better in certain circumstances. The only reason IPython is not killed
when you go to lunch is because of websockets.

It doesn't matter if checkpointing occurs regularly if the container is
killed. Agreed. Don't want to persist the checkpoints to the user's history.
15. okt. 2015 9.33 a.m. skrev "Björn Grüning" notifications@github.com:

**Comment by bgruening** [comment]
@erasche just saying that this is important to me and this makes a huge difference. What about setting up a Docker container with the new proxy and seeking for help in testing. After this we merge it into main and get even more testers. ETA for main late November?

**Comment by hexylena** [comment]
@bgruening sorry, yeah, I came off a bit aggressively, my apologies. 

Definitely, let's set up a container and test it. I don't mind when this gets in, just that we move to something cleaner that what is currently in existence.
